src/sonic-yang-models/yang-models/sonic-telemetry_client.yang

```diff
@@ -86,33 +86,26 @@ module sonic-telemetry_client {
                 }
             }
 
-            list TELEMETRY_CLIENT_DS_LIST {
+            list TELEMETRY_CLIENT_LIST {
```

Comment:
Now fields from both DestinationGroup and Subscription keys mixed up. For example, earlier the field "dst_addr" was valid for DestinationGroup key only. Now it will be accepted for Subscription keys too. Better to segregate them using choice/case statements and guard them using a when statement. Like:
```
list TELEMETRY_CLIENT_LIST {
  key "prefix name";
  leaf prefix { ... }
  leaf name { ... }
  choice key_type {
    case destination-group {
      when "../prefix = 'DestinationGroup'";
      leaf dst_addr { ... }
    }
    case subscription {
      when "../prefix = 'Subscription'";
      // fields from old TELEMETRY_CLIENT_SUB_LIST 
    }
  }
```

Reply:
@sachinholla Hi, it seems a bug in `choice` statement. When using `choice` in any of yang model other than `sonic-acl.yang`, it will fail the yang validation.
I submitted the issue. Please check detail here: https://github.com/sonic-net/sonic-buildimage/issues/17518

I suggest to merge this PR first. Then the field leaf can be enhanced after `choice` issue being fixed.

Reply:
Okay.. let us go ahead since the author of this yang already approved